coalib/io/Directory.py

```diff
@@ -0,0 +1,89 @@
+import os
+
+from coalib.parsing.Globbing import relative_recursive_glob
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq('path', 'timestamp')
+class Directory:
+    """
+    The ``Directory`` class acts as an interface to directories and
+    provides useful information about them such as:
+
+        * The directory path.
+        * The parent directory path.
+        * The last modified timestamp of the directory.
+        * The children contained inside a directory.
+
+    >>> import os
+    >>> d = Directory('tests/io/DirectoryTestDir/')
+
+    Get the number of files and sub-directories at the top level:
+
+    >>> len(d.get_children())
+    3
+
+    Get the number of all the files and sub-directories recursively:
+    >>> len(d.get_children_recursively())
+    5
+
+    Get the path of the ``Directory`` object:
+
+    >>> os.path.basename(d.path).endswith('DirectoryTestDir')
+    True
+
+    Get the parent directory:
+
+    >>> os.path.basename(d.parent).endswith('io')
```

Comment:
maybe `'tests/io'`

Reply:
cant do that or it would fail on windows